_posts/2019/2019-07-24-how-to-transpile-node-modules-with-babel-and-webpack-in-a-monorepo.md

```diff
@@ -0,0 +1,140 @@
+
+Essentially we changed this:
+
+```
+...
+  exclude: /node_modules/,
+...
+```
+
+to this:
+
+```
+...
+  exclude: /node_modules\/?!name-of-untranspiled-module.*/,
+...
+```
+
+## Why this didn't work...
+
+We should have expected this from the onset, but upon rebuilding the project we errored again on IE11, because _another module_ was also not transpiled!
+Seeing as it would be a laborious process to add every untranspiled module to our regex, we decided to figure out a way to do this check programmatically.
+
+## Enter are-you-es5
+
+[are-you-es5](https://github.com/obahareth/are-you-es5){:target="_blank"} is a nifty npm package which according to its readme, is *"A package to help you find out which of your `node_modules` aren't written in ES5 so you can add them to your Webpack/Rollup/Parcel transpilation steps"*.
+It must be said that the package works entirely as advertised, and ordinarily I would recommend this for use in any project that may run on Internet Explorer, except for the following caveat: **It does not work when used in monorepos**.
+
+### What is a monorepo?
+
+A [monorepo](https://en.wikipedia.org/wiki/Monorepo){:target="_blank"}, short for *monolithic repository*, is  a software development strategy where code for several projects is stored in a single repository.
+The advantages of using a monorepo structure are numerous and include simplified dependency management, streamlining of large scale refactoring processes and ease of code reuse among others.
+Despite all the vaunted advantages, a glaring flaw of the monorepo project structure which affected us in this case is the fact that it complicates build and version control processes.
+There aren't many tools that help overcome the version control issues, and this has led to companies such as Facebook and Microsoft contributing heavily to or forking their own versions of their preferred version control software, or just building theirs as in the case of Google.
+Build problems are in a similar situation, so much so that Facebook and Google also developed their own build software.
+
+We are fortunate to have [Lerna](https://github.com/lerna/lerna){:target="_blank"}, a tool which optimizes the workflow around monorepos with git and npm, and for this reason fits our purposes quite nicely.
+The way Lerna sets up monorepos, most `node_modules` folders end up in the root folder, but are referenced in the `package.json` of their respective packages.
+`are-you-es5` is currently not configured to work with such a project structure.
+
+## Enter depoulo/are-you-es5
+
+[Paolo Priotto](https://github.com/depoulo){:target="_blank"} forked the `are-you-es5` repository and changed it so instead of hardcoding a path to the `node_modules` directory, `require.resolve()` is used to generate an absolute path to each package's directory.
```

Comment:
```suggestion
[Paolo Priotto](https://github.com/depoulo){:target="_blank"} forked the `are-you-es5` repository and changed it so that instead of hardcoding a path to the `node_modules` directory, `require.resolve()` is used to generate an absolute path to each package's directory.
```